### ISSUE TITLE
Http requests throw XMLHttpRequest error

### ISSUE THREAD

**Issue by Nishanth1311** [flutter/flutter#85985]
A simple http request, an error was reported on flutter web, The request succeeds in python/JS when I run it separately.
But it always fails in flutter.

## Steps to Reproduce

The API used is a simple helloworld API from firebase cloud.
    **_String url = 'https://us-central1-app1fb.cloudfunctions.net/helloWorld';
    String rsp;
    final response = http.get(url).then((value) => {rsp = value.body});
    print(rsp);_**

**Expected results:** 
Hello From Firebase!

**Actual results:** 
Error: XMLHttpRequest error.
    dart-sdk/lib/_internal/js_dev_runtime/patch/core_patch.dart 909:28                get current
packages/http/src/browser_client.dart 84:22                                       <fn>
dart-sdk/lib/async/zone.dart 1613:54                                              runUnary
dart-sdk/lib/async/future_impl.dart 155:18                                        handleValue
dart-sdk/lib/async/future_impl.dart 707:44                                        handleValueCallback
dart-sdk/lib/async/future_impl.dart 736:13                                        _propagateToListeners
dart-sdk/lib/async/future_impl.dart 533:7                                         [_complete]
dart-sdk/lib/async/stream_pipe.dart 61:11                                         _cancelAndValue
dart-sdk/lib/async/stream.dart 1219:7                                             <fn>
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 324:14  _checkAndCall
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 329:39  dcall
dart-sdk/lib/html/dart2js/html_dart2js.dart 37307:58                              <fn>

    at Object.createErrorWithStack (http://localhost:37569/dart_sdk.js:5362:12)
    at Object._rethrow (http://localhost:37569/dart_sdk.js:39548:16)
    at async._AsyncCallbackEntry.new.callback (http://localhost:37569/dart_sdk.js:39542:13)
    at Object._microtaskLoop (http://localhost:37569/dart_sdk.js:39374:13)
    at _startMicrotaskLoop (http://localhost:37569/dart_sdk.js:39380:13)
    at http://localhost:37569/dart_sdk.js:34887:9

<details>
  <summary>Logs</summary>

```
[  +41 ms] executing: uname -m
[  +20 ms] Exit code 0 from: uname -m
[        ] x86_64
[   +3 ms] executing: [/home/nishanth/snap/flutter/common/flutter/] git -c log.showSignature=false log -n 1 --pretty=format:%H
[   +3 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[        ] d79295af24c3ed621c33713ecda14ad196fd9c31
[        ] executing: [/home/nishanth/snap/flutter/common/flutter/] git tag --points-at d79295af24c3ed621c33713ecda14ad196fd9c31
[  +12 ms] Exit code 0 from: git tag --points-at d79295af24c3ed621c33713ecda14ad196fd9c31
[        ] 2.2.2
[   +6 ms] executing: [/home/nishanth/snap/flutter/common/flutter/] git rev-parse --abbrev-ref --symbolic @{u}
[   +4 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{u}
[        ] origin/stable
[        ] executing: [/home/nishanth/snap/flutter/common/flutter/] git ls-remote --get-url origin
[   +3 ms] Exit code 0 from: git ls-remote --get-url origin
[        ] https://github.com/flutter/flutter.git
[  +48 ms] Unable to locate an Android SDK.
[   +4 ms] executing: [/home/nishanth/snap/flutter/common/flutter/] git rev-parse --abbrev-ref HEAD
[   +3 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[        ] stable
[  +33 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[   +1 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[  +40 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[  +42 ms] Skipping pub get: version match.
[  +63 ms] Generating /home/nishanth/coding/random/newapp/android/app/src/main/java/io/flutter/plugins/GeneratedPluginRegistrant.java
[  +94 ms] Launching lib/main.dart on Chrome in debug mode...
[  +62 ms] Updating assets
[  +43 ms] Waiting for connection from debug service on Chrome...
[  +57 ms] <- reset
[   +3 ms] /home/nishanth/snap/flutter/common/flutter/bin/cache/dart-sdk/bin/dart --disable-dart-dev
/home/nishanth/snap/flutter/common/flutter/bin/cache/artifacts/engine/linux-x64/frontend_server.dart.snapshot --sdk-root /home/nishanth/snap/flutter/common/flutter/bin/cache/flutter_web_sdk/
--incremental --target=dartdevc --debugger-module-names --experimental-emit-debug-metadata -DFLUTTER_WEB_AUTO_DETECT=true --output-dill /tmp/flutter_tools.QOPYCS/flutter_tool.XJGYWK/app.dill
--libraries-spec file:///home/nishanth/snap/flutter/common/flutter/bin/cache/flutter_web_sdk/libraries.json --packages /home/nishanth/coding/random/newapp/.dart_tool/package_config.json
-Ddart.vm.profile=false -Ddart.vm.product=false --enable-asserts --track-widget-creation --filesystem-root /tmp/flutter_tools.QOPYCS/flutter_tools.AOOTXW --filesystem-scheme org-dartlang-app
--initialize-from-dill build/c7922d95bf4a2462b75c84a97c312edb.cache.dill.track.dill --platform file:///home/nishanth/snap/flutter/common/flutter/bin/cache/flutter_web_sdk/kernel/flutter_ddc_sdk.dill
--no-sound-null-safety
[   +7 ms] <- compile org-dartlang-app:/web_entrypoint.dart
[+11500 ms] Waiting for connection from debug service on Chrome... (completed in 11.6s)
[        ] Synced 30.6MB.
[        ] <- accept
[        ] Caching compiled dill
[  +56 ms] Using Google Chrome 91.0.4472.114 

[ +240 ms] [CHROME]:
[        ] [CHROME]:DevTools listening on ws://127.0.0.1:46719/devtools/browser/20596980-211a-4b2d-beb8-c991f5a1db4c
[ +155 ms] DwdsInjector: Received request for entrypoint at http://localhost:35511/main_module.bootstrap.js
[   +6 ms] MetadataProvider: Loading debug metadata...
[  +10 ms] MetadataProvider: Loaded debug metadata (no sound null safety)
[   +6 ms] DwdsInjector: Injected debugging metadata for entrypoint at http://localhost:35511/main_module.bootstrap.js
[+1091 ms] ChromeProxyService: Initializing expression compiler for main_module.bootstrap.js with sound null safety: false
[  +43 ms] DevHandler: Debug service listening on ws://127.0.0.1:46083/XRtwr-Ne5sQ=/ws

[  +14 ms] This app is linked to the debug service: ws://127.0.0.1:46083/XRtwr-Ne5sQ=/ws
[   +4 ms] Debug service listening on ws://127.0.0.1:46083/XRtwr-Ne5sQ=/ws
[        ] Running with unsound null safety
[        ] For more information see https://dart.dev/null-safety/unsound-null-safety
[   +1 ms] 🔥  To hot restart changes while running, press "r" or "R".
[        ] For a more detailed help message, press "h". To quit, press "q".
[+6376 ms] null
[ +313 ms] null
[ +391 ms] Error: XMLHttpRequest error.
                        dart-sdk/lib/_internal/js_dev_runtime/patch/core_patch.dart 909:28                get current
                    packages/http/src/browser_client.dart 84:22                                       <fn>
                    dart-sdk/lib/async/zone.dart 1613:54                                              runUnary
                    dart-sdk/lib/async/future_impl.dart 155:18                                        handleValue
                    dart-sdk/lib/async/future_impl.dart 707:44                                        handleValueCallback
                    dart-sdk/lib/async/future_impl.dart 736:13                                        _propagateToListeners
                    dart-sdk/lib/async/future_impl.dart 533:7                                         [_complete]
                    dart-sdk/lib/async/stream_pipe.dart 61:11                                         _cancelAndValue
                    dart-sdk/lib/async/stream.dart 1219:7                                             <fn>
                    dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 324:14  _checkAndCall
                    dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 329:39  dcall
                    dart-sdk/lib/html/dart2js/html_dart2js.dart 37307:58                              <fn>
                    
                    
                        at Object.createErrorWithStack (http://localhost:35511/dart_sdk.js:5362:12)
                        at Object._rethrow (http://localhost:35511/dart_sdk.js:39548:16)
                        at async._AsyncCallbackEntry.new.callback (http://localhost:35511/dart_sdk.js:39542:13)
                        at Object._microtaskLoop (http://localhost:35511/dart_sdk.js:39374:13)
                        at _startMicrotaskLoop (http://localhost:35511/dart_sdk.js:39380:13)
                        at http://localhost:35511/dart_sdk.js:34887:9
[   +2 ms] null
```

```
flutter analyze
Analyzing newapp...                                                     
No issues found! (ran in 0.7s)
```

```
[✓] Flutter (Channel stable, 2.2.2, on Linux, locale en_IN)
    • Flutter version 2.2.2 at /home/nishanth/snap/flutter/common/flutter
    • Framework revision d79295af24 (4 weeks ago), 2021-06-11 08:56:01 -0700
    • Engine revision 91c9fc8fe0
    • Dart version 2.13.3

[✗] Android toolchain - develop for Android devices
    ✗ Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.dev/docs/get-started/install/linux#android-setup for detailed instructions).
      If the Android SDK has been installed to a custom location, please use
      `flutter config --android-sdk` to update to that location.


[✓] Chrome - develop for the web
    • Chrome at google-chrome

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/linux#android-setup for detailed instructions).

[✓] Connected device (1 available)
    • Chrome (web) • chrome • web-javascript • Google Chrome 91.0.4472.114

! Doctor found issues in 2 categories.
```

</details>

**Comment by thunderstorm010** [comment]
You are printing before the request completes. Either use await or put print in the .then block. Check the example on https://pub.dev/packages/http

**Comment by darshankawar** [comment]
@Nishanth1311 
Check if above solution helps. If not, I think this is related to CORS over which flutter has no control. You may need to enable it in the backend you are using.

Closing as this isn't a flutter issue. If you disagree, write in comments and I'll reopen it.

**Comment by Nishanth1311** [comment]
@darshankawar I understand that I am printing it incorrectly. I've attached the wrong code here.
But I did try printing the result in the 'then' statement.

**_String url = 'https://us-central1-app1fb.cloudfunctions.net/helloWorld';
String rsp;
final response = http.get(url).then((value) => {rsp = value.body, print(rsp)});_**

But the above code does not work as well.
I can confirm that the HTTP calls are working well in node/python.
Its just failing in flutter.
